database/engine/metadata_log/metalogpluginsd.c

```diff
@@ -131,22 +139,33 @@ PARSER_RC metalog_pluginsd_context_action(void *user, uuid_t *uuid)
                     chart_char_guid = (uuid_t *)(object + 16);
 
                     ret = find_object_by_guid(chart_char_guid, id_str, RRD_ID_LENGTH_MAX + 1);
-                    fatal_assert(GUID_TYPE_CHAR == ret);
-                    ((PARSER_USER_OBJECT *)user)->st = rrdset_find(host, id_str);
+                    if (unlikely(GUID_TYPE_CHAR != ret)) {
+                        error_with_guid(uuid, "Failed to find valid chart name");
+                        ((PARSER_USER_OBJECT *)user)->st = NULL;
+                    }
+                    else
+                        ((PARSER_USER_OBJECT *)user)->st = rrdset_find(host, id_str);
                     break;
                 case GUID_TYPE_DIMENSION:
                     chart_guid = (uuid_t *)(object + 16);
 
                     ret = find_object_by_guid(chart_guid, chart_object, 33);
-                    fatal_assert(GUID_TYPE_CHART == ret);
-                    chart_char_guid = (uuid_t *)(chart_object + 16);
+                    if (unlikely(GUID_TYPE_CHART != ret)) {
+                        error_with_guid(uuid, "Failed to find valid chart");
+                        ((PARSER_USER_OBJECT *)user)->st = NULL;
+                        break;
+                    }
+                    chart_char_guid = (uuid_t *)(object + 16);
 
                     ret = find_object_by_guid(chart_char_guid, id_str, RRD_ID_LENGTH_MAX + 1);
-                    fatal_assert(GUID_TYPE_CHAR == ret);
-                    ((PARSER_USER_OBJECT *)user)->st = rrdset_find(host, id_str);
+                    if (unlikely(GUID_TYPE_CHAR != ret)) {
+                        error_with_guid(uuid, "Failed to find valid chart name");
+                        ((PARSER_USER_OBJECT *)user)->st = NULL;
+                    }
+                    else
+                        ((PARSER_USER_OBJECT *)user)->st = rrdset_find(host, id_str);
                     break;
                 default:
-                    fatal_assert(0);
```

Comment:
This should be logged as it indicates that the log is either corrupt or from a newer version than the agent.